### PR TITLE
Adding support for detailed monitoring of instances

### PR DESCRIPTION
**Pull request by joshgarnett**
This resolves issue #1246

### CODE REVIEW

File: builtin/providers/aws/resource_aws_instance.go [hunk 1]
```diff
@@ -463,6 +469,12 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
 		d.Set("subnet_id", instance.SubnetID)
 	}
 	d.Set("ebs_optimized", instance.EBSOptimized)
+
+	if instance.Monitoring != nil && instance.Monitoring.State != nil {
```
Comment (mitchellh) [inline builtin/providers/aws/resource_aws_instance.go]: Nice nil checks here

File: builtin/providers/aws/resource_aws_instance.go [hunk 2]
```diff
@@ -463,6 +469,12 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
...
+		monitoring_state := *instance.Monitoring.State
```
Comment (mitchellh) [inline builtin/providers/aws/resource_aws_instance.go]: It is more idiomatic to use `monitoringState`, but I can fix that up.

**Comment by mitchellh** [comment]
LGTM

File: builtin/providers/aws/resource_aws_instance.go (hunk 2 above)
Comment (joshgarnett) [inline builtin/providers/aws/resource_aws_instance.go]: Ahh yes, sorry I've been bouncing between ruby & go lately.

**Pull request merged**